src/views/ModelPlan/TaskList/Basics/Milestones/index.tsx

```diff
@@ -0,0 +1,634 @@
+          highLevelNote: formikValues.highLevelNote,
+          phasedIn: formikValues.phasedIn,
+          phasedInNote: formikValues.phasedInNote
+        }
+      }
+    })
+      .then(response => {
+        if (!response?.errors) {
+          if (redirect === 'task-list') {
+            history.push(`/models/${modelId}/task-list`);
+          }
+        }
+      })
+      .catch(errors => {
+        formikRef?.current?.setErrors(errors);
+      });
+  };
+
+  return (
+    <MainContent className="margin-bottom-5">
+      <GridContainer>
+        <Grid desktop={{ col: 12 }}>
+          <BreadcrumbBar variant="wrap">
+            <Breadcrumb>
+              <BreadcrumbLink asCustom={Link} to="/">
+                <span>{h('home')}</span>
+              </BreadcrumbLink>
+            </Breadcrumb>
+            <Breadcrumb>
+              <BreadcrumbLink
+                asCustom={Link}
+                to={`/models/${modelId}/task-list/`}
+              >
+                <span>{h('tasklistBreadcrumb')}</span>
+              </BreadcrumbLink>
+            </Breadcrumb>
+            <Breadcrumb current>{t('breadcrumb')}</Breadcrumb>
+          </BreadcrumbBar>
+          <PageHeading className="margin-top-4 margin-bottom-1">
+            {t('heading')}
+          </PageHeading>
+          <p
+            className="margin-top-0 margin-bottom-1 font-body-lg"
+            data-testid="model-plan-name"
+          >
+            <Trans i18nKey="modelPlanTaskList:subheading">
+              indexZero {modelName} indexTwo
+            </Trans>
+          </p>
+          <p className="margin-bottom-2 font-body-md line-height-sans-4">
+            {t('helpText')}
+          </p>
+
+          <Formik
+            initialValues={initialValues}
+            onSubmit={values => {
+              handleFormSubmit(values, 'task-list');
+            }}
+            enableReinitialize
+            validationSchema={planBasicsSchema.pageThreeSchema}
+            validateOnBlur={false}
+            validateOnChange={false}
+            validateOnMount={false}
+            innerRef={formikRef}
+          >
+            {(formikProps: FormikProps<PlanBasicsMilestoneTypes>) => {
+              const {
+                dirty,
+                errors,
+                handleSubmit,
+                isValid,
+                setErrors,
+                setFieldError,
+                setFieldValue,
+                validateField,
+                values
+              } = formikProps;
+              const flatErrors = flattenErrors(errors);
+              const handleOnchange = (e: any, field: string) => {
+                if (e === '') {
+                  return;
+                }
+                try {
+                  setErrors({});
+                  setFieldValue(field, new Date(e).toISOString());
+                  validateField(field);
+                } catch (error) {
+                  setFieldError(field, 'Please use a valid date format');
```

Comment:
Honestly I think this should existing in a validation schema, and on not onchange/every key stroke.  The validation should happen once on submit.  I'm able to submit the form with errors if I type something incorrectly, then type something incorrectly in another box, the first validation error gets cleared.  I think autosave might prove to be tricky here, so feel free to leave out autosave on this page due to its complexity with dates.